src/main/kotlin/com/openlattice/search/PersistentSearchMessengerTask.kt

```diff
@@ -198,33 +198,45 @@ class PersistentSearchMessengerTask : HazelcastFixedRateTask<PersistentSearchMes
     }
 
     override fun runTask() {
+
         logger.info("Loading new writes for persistent searches and sending alerts")
 
         val dependencies = getDependency()
-        val totalSearchesUpdated = dependencies.hds.connection.use { outConnection ->
-            outConnection.prepareStatement(updateLastReadSql()).use { ps ->
-                PostgresIterable(Supplier<StatementHolder> {
-                    val connection = dependencies.hds.connection
-                    connection.autoCommit = false
-                    val stmt = connection.createStatement()
-                    stmt.fetchSize = 32000
-                    val rs = stmt.executeQuery(LOAD_ACTIVE_ALERTS_SQL)
-                    connection.autoCommit = true
-                    StatementHolder(connection, stmt, rs)
-                }, Function<ResultSet, Pair<AclKey, PersistentSearch>> {
-                    ResultSetAdapters.aclKey(it) to ResultSetAdapters.persistentSearch(it)
-                }).forEach { (aclKey, search) ->
-                    val latestRead = findNewWritesForAlert(aclKey, search)
-                    if (latestRead != null) {
-                        ps.setObject(1, latestRead)
-                        ps.setObject(2, search.id)
-                        if (logger.isDebugEnabled) {
-                            logger.debug("Updating last read for $aclKey and $search")
-                        } else {
-                            logger.info("Updating last read for $aclKey and ${search.id}")
-                        }
-                        ps.addBatch()
+
+        val persistentSearchesById = PostgresIterable(Supplier<StatementHolder> {
```

Comment:
Minor thing, but can you pls use `BasePostgresIterable` and `StatementHolderSupplier`?